src/collections/PVProductionHistory.ts

```diff
@@ -57,10 +60,232 @@ export const PVProductionHistory: CollectionConfig = {
           required: false,
           admin: {
             description:
-              'Estimated loss of pv production data due to weather conditions as percentag (0-100%)',
+              'Estimated loss of pv production data due to weather conditions as percentage (0-100%)',
           },
         },
       ],
     },
   ],
 }
+
+export type PVProductionImport = {
+  measured_production: MeasuredProductionData[]
+}
+
+export type MeasuredProductionData = {
+  from: string
+  to: string
+  production: number
+}
+
+export type ParsedMeasuredProductionData = {
+  from: Date
+  to: Date
+  production: number
+}
+
+export async function importPVProductionData(
+  installationId: number,
+  req: PayloadRequest,
+): Promise<Response> {
+  await addDataAndFileToRequest(req)
+  if (req.json) {
+    console.debug('import data for installation', installationId)
+    const data: PVProductionImport = await req.json()
+
+    if (!data || !data.measured_production || data.measured_production.length === 0) {
+      return Response.json(
+        { error: 'At least one measured production entry required' },
+        { status: 400 },
+      )
+    }
+
+    const parseEntries = data.measured_production.map((entry) => {
+      const parsed: ParsedMeasuredProductionData = {
+        from: new Date(entry.from),
+        to: new Date(entry.to),
+        production: entry.production,
+      }
+      return parsed
+    })
+
+    parseEntries.forEach((entry, index) => {
+      if (entry.from.getTime() >= entry.to.getTime()) {
+        return Response.json(
+          { error: `Import validation failed on row [${index}]: invalid date range` },
+          { status: 400 },
+        )
+      }
+    })
+
+    const earliestFrom = parseEntries
+      .map((x) => x.from)
+      .sort((a, b) => a.getTime() - b.getTime())[0]
+    const latestTo = parseEntries.map((x) => x.to).sort((a, b) => b.getTime() - a.getTime())[0]
+
+    const installation = await req.payload.findByID({
+      id: installationId,
+      collection: 'installations',
+    })
+
+    // TODO: enrich with external sources
```

Comment:
I allow you this since this is only a MVP ;)

Reply:
as we are in development phase of the MVP I want to merge as early as possible which doesn't mean all the functionality if a feature is implemented in a PR. Therefore it makes sense to enrich with TODO's we want to tackle within or after finishing the MVP.